arangod/RestHandler/RestCollectionHandler.cpp

```diff
@@ -593,28 +594,42 @@ RestStatus RestCollectionHandler::handleCommandPut() {
         }));
 
   } else if (sub == "upgrade") {
-    return waitForFuture(
-        methods::Collections::upgrade(_vocbase, *coll).thenValue([this, coll](Result&& res) {
-          if (res.fail()) {
-            generateTransactionError(coll->name(), res, "");
-            return;
-          }
+    using UpgradeState = LogicalCollection::UpgradeStatus::State;
```

Comment:
This looks like a step backward to synchronous operation. Maybe the future business was built after you started the PR? Or is there a reason to go back to synchronous operation?

Reply:
Yes, there's not really any asynchronous behavior contained. The only underlying method that is a target for asynchronous implementation is `ClusterMethods::upgradeOnCoordinator`, as it interacts with the agency; however, the current implementation for that is also synchcronous, so it doesn't really make sense to wrap everything in futures.